packages/block-editor/src/components/editor-styles/index.js

```diff
@@ -66,7 +66,56 @@ function useDarkThemeBodyClassName( styles ) {
 	);
 }
 
-export default function EditorStyles( { styles } ) {
+function useParsedAssets( html ) {
+	return useMemo( () => {
+		const doc = document.implementation.createHTMLDocument( '' );
+		doc.body.innerHTML = html;
+		return Array.from( doc.body.children );
+	}, [ html ] );
+}
+
+function PrefixedStyle( { tagName, href, id, rel, media, textContent } ) {
+	const TagName = tagName.toLowerCase();
+
+	function onLoad( event ) {
```

Comment:
This technique is cool, though there's a noticeable delay in the styles being applied. Presumably, because we add the stylesheet in the client after all things have been already loaded and the `onLoad` event happens even after.

Thinking about how to improve this:

- What if we convert all styles (both `link` and `style`) to embedded styles (`style`) and prefixed them just here, so we don't wait to `onLoad` to do it?
- Alternatively, would it make sense to add the stylesheets in the server (no passing them through `__unstableResolvedContentStyles`) and attach to its `onLoad` event a public function that takes care of prefixing them?

Reply:
The good think here is that we don't have to do any CSS parsing. I talked with @mcsf about the delay. I wonder if we could block the original CSS from applying, maybe by emptying and later refilling it. 🤔
It would be good if we don't have to do any parsing. I was hoping there would be some API to scope link and style tags to a given selector, but that doesn't exist apparently.

Reply:
This would be ideal: https://css-tricks.com/saving-the-day-with-scoped-css/

Reply:
@oandregal Could you remind me where you see the delay?

I now removed the prefixing from the iframes since it's not needed there. Only the "old" non iframe editors will need prefixing.

Reply:
I was able to repro with a big post (Gutenberg demo). It's more noticeable the 1st time the post is loaded and if you disable the cache. Also by throtling the network and/or the CPU it becomes more visible.

Reply:
What is more visible? What do you see? Could you record it?